geoips/commandline/geoips_test.py

```diff
@@ -0,0 +1,212 @@
+        Where:
+            - <package_name> is any GeoIPS package that is installed and recognized by
+              the GeoIPS Library
+            - <script_name> is the name of the bash script being tested
+            - '--integration' represents whether or not this is an 'integration' test
+        """
+        self.subcommand_parser.add_argument(
+            "script_name",
+            type=str,
+            help="GeoIPS Script to be tested",
+        )
+        self.subcommand_parser.add_argument(
+            "--package_name",
+            "-p",
+            type=str,
+            default="geoips",
+            choices=self.plugin_packages,
+            help="GeoIPS Package containing the script to be tested",
+        )
+        self.subcommand_parser.add_argument(
+            "--integration",
+            default=False,
+            action="store_true",
+            help="Whether or not the script is an integration test.",
+        )
+
+    def __call__(self, args):
+        """Run the provided test script based off the arguments provided.
+
+        This will run the provided test script, locating such script based on whether
+        or not this was an integration-based test. If it wasn't, we'll look through the
+        corresponding <package_name>/tests/scripts/ directory for the correct test
+        script.
+
+        Parameters
+        ----------
+        args: Argparse Namespace()
+            - The list argument namespace to parse through
+        """
+        package_name = args.package_name
+        script_name = args.script_name
+        is_integration_test = args.integration
+
+        if is_integration_test:
+            if package_name != "geoips":
+                # Raise an argparse error which states only geoips supports integration
+                # testing
+                err_str = "Only package 'geoips' has integration tests. Package "
+                err_str += f"'{package_name}' doesn't have those tests. Try again."
+                self.subcommand_parser.error(err_str)
+            dir_name = "integration_tests"
+        else:
+            dir_name = "scripts"
+
+        test_dir = str(resources.files(package_name) / f"../tests/{dir_name}")
+        fnames = [basename(fpath) for fpath in glob(f"{test_dir}/*.sh")]
+
+        if script_name not in fnames:
+            # Raise an argparse error which states that file doesn't exist within the
+            # specified directory
+            str_fnames = ",\n".join(fnames)
+            err_str = f"Script '{script_name}' doesn't exist within '{test_dir}/'. "
+            err_str += "Please select a valid script from this directory listing:\n"
+            err_str += str_fnames
+            self.subcommand_parser.error(err_str)
+
+        script_path = f"{test_dir}/{script_name}"
+        # Run the corresponding test
+        call(["bash", script_path], shell=False)
+
+
+class GeoipsTestLinting(GeoipsExecutableCommand):
```

Comment:
This calls the linters in the same manner as is done in the integration tests. The only difference is that we iterate over each linter / formatter rather than calling ``check_code.sh`` via ``./check_code all <repo_path>``. ``all`` additionally runs unit tests, which we don't want in this case.

Reply:
So for ``geoips test linting -p geoips``, the command calls would look like this:
* ``bash /path/to/geoips/tests/utils/check_code.sh bandit /path/to/geoips/.``
* ``bash /path/to/geoips/tests/utils/check_code.sh black /path/to/geoips/.``
* ``bash /path/to/geoips/tests/utils/check_code.sh flake8 /path/to/geoips/.``

Reply:
What additional CLI arguments were you thinking @jsolbrig?

Reply:
I think this may be part of a larger issue. We need to address how we set the linter options for GeoIPS in general. The options can currently be found in geoips/.config, but that isn't a standard location. I think there is already an issue for this. I'll see if I can find that and reference this conversation there.